packages/aws-cdk/lib/api/aws-auth/sdk-provider.ts

```diff
@@ -575,3 +592,21 @@ function fmtObtainedCredentials(
       return msg.join('');
   }
 }
+
+/**
+ * Instantiate an SDK for context providers. This function ensures that all
+ * lookup assume role options are used when context providers perform lookups.
+ */
+export async function initContextProviderSdk(aws: SdkProvider, options: cxschema.ContextLookupRoleOptions): Promise<ISDK> {
```

Comment:
@rix0rrr Can you have a look at this (and how its being used)? I added it since your last review to fix the lookup issues.